### PR TITLE
Introduce Size type

### PR DESCRIPTION
**Pull request by kkaefer**
We frequently specify sizes that have a width and a height and should introduce a size type that encodes this instead of using `std::array<uint16_t, 2>` in those places.

/cc @jfirebaugh

### CODE REVIEW

File: include/mbgl/map/map.hpp [hunk 1]
```diff
@@ -136,9 +137,9 @@ class Map : private util::noncopyable {
     ViewportMode getViewportMode() const;
 
     // Size
-    void setSize(const std::array<uint16_t, 2>&);
-    uint16_t getWidth() const;
-    uint16_t getHeight() const;
+    void setSize(Size);
+    uint32_t getWidth() const;
```
Comment (jfirebaugh) [inline include/mbgl/map/map.hpp]: We should combine `getWidth()` and `getHeight()` into a single `getSize()`.

File: include/mbgl/sprite/sprite_image.hpp [hunk 2]
```diff
@@ -22,8 +22,8 @@ class SpriteImage : private util::noncopyable {
     // Whether this image should be interpreted as a signed distance field icon.
     const bool sdf;
 
-    float getWidth() const { return image.width / pixelRatio; }
-    float getHeight() const { return image.height / pixelRatio; }
+    float getWidth() const { return image.size.width / pixelRatio; }
```
Comment (jfirebaugh) [inline include/mbgl/sprite/sprite_image.hpp]: Same here.

File: platform/default/offscreen_view.cpp [hunk 3]
```diff
@@ -21,17 +21,17 @@ void OffscreenView::bind() {
         context.bindFramebuffer = framebuffer->framebuffer;
     }
 
-    context.viewport = { 0, 0, size[0], size[1] };
+    context.viewport = { 0, 0, size.width, size.height };
```
Comment (jfirebaugh) [inline platform/default/offscreen_view.cpp]: Should be able to do `context.viewport = { 0, 0, size }`.

File: include/mbgl/sprite/sprite_image.hpp (hunk 2 above)
Comment (kkaefer) [inline include/mbgl/sprite/sprite_image.hpp]: In this particular case, we're using `float`s to represent the width/height, but our `Size` type uses integers, so we should keep this.

**Pull request merged**